ui/app/components/pages/mobile-sync/index.js

```diff
@@ -0,0 +1,387 @@
+const { Component } = require('react')
+const { connect } = require('react-redux')
+const PropTypes = require('prop-types')
+const h = require('react-hyperscript')
+const classnames = require('classnames')
+const PubNub = require('pubnub')
+
+const { requestRevealSeedWords, fetchInfoToSync } = require('../../../actions')
+const { DEFAULT_ROUTE } = require('../../../routes')
+const actions = require('../../../actions')
+
+const qrCode = require('qrcode-generator')
+
+import Button from '../../button'
+import LoadingScreen from '../../loading-screen'
+
+const PASSWORD_PROMPT_SCREEN = 'PASSWORD_PROMPT_SCREEN'
+const REVEAL_SEED_SCREEN = 'REVEAL_SEED_SCREEN'
+
+class MobileSyncPage extends Component {
+  static propTypes = {
+    history: PropTypes.object,
+    selectedAddress: PropTypes.string,
+    displayWarning: PropTypes.func,
+    fetchInfoToSync: PropTypes.func,
+    requestRevealSeedWords: PropTypes.func,
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      screen: PASSWORD_PROMPT_SCREEN,
+      password: '',
+      seedWords: null,
+      error: null,
+      syncing: false,
+      completed: false,
+    }
+
+    this.syncing = false
+  }
+
+  componentDidMount () {
+    const passwordBox = document.getElementById('password-box')
+    if (passwordBox) {
+      passwordBox.focus()
+    }
+  }
+
+  handleSubmit (event) {
+    event.preventDefault()
+    this.setState({ seedWords: null, error: null })
+    this.props.requestRevealSeedWords(this.state.password)
+      .then(seedWords => {
+        this.generateCipherKeyAndChannelName()
+        this.setState({ seedWords, screen: REVEAL_SEED_SCREEN })
+        this.initWebsockets()
+      })
+      .catch(error => this.setState({ error: error.message }))
+  }
+
+  generateCipherKeyAndChannelName () {
+    this.cipherKey = `${this.props.selectedAddress.substr(-4)}-${PubNub.generateUUID()}`
+    this.channelName = `mm-${PubNub.generateUUID()}`
+  }
+
+  initWebsockets () {
+    this.pubnub = new PubNub({
+      subscribeKey: 'sub-c-30b2ba04-c37e-11e8-bd78-d63445bede87',
+      publishKey: 'pub-c-d40e77d5-5cd3-4ca2-82eb-792a1f4573db',
```

Comment:
We gotta rotate keys before merging this in, move the keys into CircleCI env variables and consume it here via process.env.PUBNUB_SUB_KEY and process.env.PUBNUB_PUB_KEY